analytics/app/src/main/java/com/google/firebase/quickstart/analytics/java/MainActivity.java

```diff
@@ -104,28 +106,31 @@ protected void onCreate(Bundle savedInstanceState) {
             setUserFavoriteFood(userFavoriteFood);
         }
 
-
         // Create the adapter that will return a fragment for each image.
-        mImagePagerAdapter = new ImagePagerAdapter(getSupportFragmentManager(), IMAGE_INFOS);
+        mImagePagerAdapter = new ImagePagerAdapter(getSupportFragmentManager(), IMAGE_INFOS, getLifecycle());
 
         // Set up the ViewPager with the pattern adapter.
-        mViewPager = binding.viewPager;
+        mViewPager =  binding.viewPager;
```

Comment:
nit: unnecessary extra whitespace

```suggestion
        mViewPager = binding.viewPager;
```